xpk.py

```diff
@@ -1097,6 +1109,74 @@ def zone_to_region(zone) -> str:
   return zone_terms[0] + '-' + zone_terms[1]
 
 
+def check_if_service_account_exist(args):
+  """Check if a service account with the given name exist in the project.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    True if service account exist, False otherwise.
+  """
+  command = (
+    f'gcloud iam service-accounts describe {args.project}-xpk-sa@{args.project}.iam.gserviceaccount.com'
+  )
+  return_code = run_command_with_updates(command, 'Service Account Describe', args, verbose=False)
+  if return_code != 0:
+    xpk_print(f'Service Account Describe request returned ERROR {return_code}')
+    return False
+  return True
+
+def create_sevice_account(args):
+  """Creates a service account in the project.
+
+  Args:
+    args: user provided arguments for running the command.
+  """
+  service_account_exist = check_if_service_account_exist(args)
+  if service_account_exist:
+    xpk_print(f'Service Account: {args.project}-xpk-sa@{args.project}.iam.gserviceaccount.com'
+              ' already exist in the project. Will not create a new one.')
+    return
+
+  command = (
+    f'gcloud iam service-accounts create {args.project}-xpk-sa \
+    --description="Service Account for XPK" \
+    --display-name="{args.project}-xpk-sa"'
+  )
+  return_code = run_command_with_updates(command, 'Service Account Create', args)
+  if return_code != 0:
+    xpk_print(f'Service Account Create request returned ERROR {return_code}')
+    xpk_print('Make sure you have Service Account Admin Role attached to your user.')
```

Comment:
To avoid breaking users who already have cluster, I have added this SA creation in `run_gke_cluster_create_command()`, which is only called when user is creating a new cluster. . Also, this PR doesn't add SA to node pools yet.

Reply:
Moved the service account creation code behind a flag which is be default turned off.